src/main/java/com/commercetools/sync/taxcategories/utils/TaxRatesUpdateActionUtils.java

```diff
@@ -0,0 +1,212 @@
+
+        try {
+            final List<UpdateAction<TaxCategory>> updateActions =
+                buildRemoveTaxRateOrTaxRateUpdateActions(
+                    oldTaxRates,
+                    newTaxRatesDrafts
+                );
+
+            updateActions.addAll(
+                buildAddTaxRateUpdateActions(
+                    oldTaxRates,
+                    newTaxRatesDrafts
+                )
+            );
+
+            return updateActions;
+        } catch (final DuplicateNameException exception) {
+            throw new BuildUpdateActionException(exception);
+        }
+    }
+
+    /**
+     * Checks if there are any tax rates which are not existing in the {@code newTaxRatesDrafts}.
+     * If there are, then "remove" tax rate update actions are built.
+     * Otherwise, if the tax rate still exists in the new draft, then compare the tax rate
+     * fields (amount, country, state, etc..), and add the computed actions to the list of update actions.
+     *
+     * @param oldTaxRates       the list of old {@link TaxRate}s.
+     * @param newTaxRatesDrafts the list of new {@link TaxRateDraft}s.
+     * @return a list of tax rate update actions if there are tax rates that are not existing
+     *         in the new draft. If the tax rate still exists in the new draft, then compare the attribute
+     *         definition fields (name, label, etc..), and add the computed actions to the list of update actions.
+     *         Otherwise, if the tax rates are identical, an empty list is returned.
+     * @throws DuplicateNameException in case there are tax rates drafts with duplicate names.
+     */
+    @Nonnull
+    private static List<UpdateAction<TaxCategory>> buildRemoveTaxRateOrTaxRateUpdateActions(
+        @Nonnull final List<TaxRate> oldTaxRates,
+        @Nonnull final List<TaxRateDraft> newTaxRatesDrafts) {
+
+        final Map<String, TaxRateDraft> newTaxRatesDraftsNameMap =
+            newTaxRatesDrafts
+                .stream().collect(
+                toMap(TaxRateDraft::getName, taxRateDraft -> taxRateDraft,
+                    (taxRateDraftA, taxRateDraftB) -> {
+                        throw new DuplicateNameException(
+                            format("Tax rates drafts have duplicated names. Duplicated tax rate "
+                                    + "name: '%s'.  Tax rates names are expected to be unique "
+                                    + "inside their tax category.",
+                                taxRateDraftA.getName()));
+                    }
+                ));
+
+        return oldTaxRates
+            .stream()
+            .map(oldTaxRate -> {
+                final String oldTaxRateName = oldTaxRate.getName();
+                final TaxRateDraft matchingNewTaxRateDraft = newTaxRatesDraftsNameMap.get(oldTaxRateName);
+                return ofNullable(matchingNewTaxRateDraft)
+                    .map(taxRateDraft -> {
+                        if (!same(oldTaxRate, taxRateDraft)) {
```

Comment:
did you try the equals method, it seems `io.sphere.sdk.models.Base` class uses [this](http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html#reflectionEquals-java.lang.Object-java.lang.Object-boolean-) apache util.
Note: unit tests are working with `equals` method.

Reply:
But it won't work here. `oldTaxRate` is of `TaxRate` class and `taxRateDraft` is of `TaxRateDraft` class
What could be simplified is `sameSubRates` method by using Objects.equals on subRates lists but if items on list will be the same but in different order it will return false and I don't think we want to generate update action in that case